qiskit/compiler/assembler.py

```diff
@@ -154,3 +162,94 @@ def assemble_circuits(circuits, run_config=None, qobj_header=None, qobj_id=None)
 
     return QasmQobj(qobj_id=qobj_id or str(uuid.uuid4()), config=userconfig,
                     experiments=experiments, header=qobj_header)
+
+
+def _replaced_with_user_los(user_lo_dict, default_los):
+    """Return user LO frequencies replaced from `default_los`.
+    Args:
+        user_lo_dict(UserLoDict): dictionary of user's LO frequencies
+        default_los(list(float)): default LO frequencies to be replaced
+    Returns:
+        List: user LO frequencies
+    """
+    res = copy.copy(default_los)
+    for channel, user_lo in user_lo_dict.items():
+        res[channel.index] = user_lo
```

Comment:
We need to check lo type before overwrite this. It overwrites both qubit and meas lo.

Reply:
@nkanazawa1989 I understand what you mean. Since `default_los` has no channel type `DriveChannel` or `MeasurementChannel`, this function cannot distinguish them and wrongly update both channels' los based on qubit index. We need to fix this bug, but I think your approach in PR #44 is too much. Adding one more channel type argument in this function will be enough. `DeviceSpecification` has a role to contain information unconfigurable for user. So it should not have user's configuration in it.

Reply:
Thank you @itoko ! What I added in [#44](https://github.com/nkanazawa1989/qiskit-terra/pull/44) is a factory method and `DeviceSpecification` is still immutable and its default configuration is protected - I think this is your intention. What I want to do in the PR is to separate `pulse.channel` and `assembler`, because `assembler` is a module dedicated to built qobj from `schedule`s.